# DATA SCIENCE CHALLENGE QUESTION #1

Question 1: Given some sample data, write a program to answer the following: click here to access the required data set

On Shopify, we have exactly 100 sneaker shops, and each of these shops sells only one model of shoe. We want to do some analysis of the average order value (AOV). When we look at orders data over a 30 day window, we naively calculate an AOV of $3145.13. Given that we know these shops are selling sneakers, a relatively affordable item, something seems wrong with our analysis. 

 - Think about what could be going wrong with our calculation. Think about a better way to evaluate this data. 
 - What metric would you report for this dataset?
 - What is its value?


In [23]:
# Importing Libraries
import pandas as pd

In [24]:
# Importing the data 
store_df = pd.read_csv("2019 Winter Data Science Intern Challenge Data Set - Sheet1.csv")
# Check the data was imported
store_df.head(5)

order_id  shop_id  user_id  order_amount  total_items payment_method  \
0         1       53      746           224            2           cash   
1         2       92      925            90            1           cash   
2         3       44      861           144            1           cash   
3         4       18      935           156            1    credit_card   
4         5       18      883           156            1    credit_card   

            created_at  
0  2017-03-13 12:36:56  
1  2017-03-03 17:38:52  
2   2017-03-14 4:23:56  
3  2017-03-26 12:43:37  
4   2017-03-01 4:35:11

In [25]:
# Checking number of rows
print(f'There are {store_df.shape[0]} rows')
# Checking number of columns
print(f'There are {store_df.shape[1]} columns')

There are 5000 rows
There are 7 columns


In [26]:
# Check data types and missing values
store_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5000 entries, 0 to 4999
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   order_id        5000 non-null   int64 
 1   shop_id         5000 non-null   int64 
 2   user_id         5000 non-null   int64 
 3   order_amount    5000 non-null   int64 
 4   total_items     5000 non-null   int64 
 5   payment_method  5000 non-null   object
 6   created_at      5000 non-null   object
dtypes: int64(5), object(2)
memory usage: 273.6+ KB


In [27]:
# Checking the desrcription of the order_amount
store_df['order_amount'].describe()

count      5000.000000
mean       3145.128000
std       41282.539349
min          90.000000
25%         163.000000
50%         284.000000
75%         390.000000
max      704000.000000
Name: order_amount, dtype: float64

In [28]:
order_max = store_df['order_amount'].max()
order_min = store_df['order_amount'].min()

print(f'The max order amount is ${order_max}')
print(f'The min order amount is ${order_min}')

The max order amount is $704000
The min order amount is $90


## Question A
There is a large distribution in the `order_amount` column, with a max of 704,000 and a min of 90. The AOV of $3145.13 was most likely calculated using the mean value, which is very susceptible to outliers in the data. When looking at the quartiles, most of the values are around a reasonable AOV for shoes. However, the max is significantly higher. This difference is another indicator that there are outliers.

In [29]:
mean_order = store_df['order_amount'].mean()
median_order = store_df['order_amount'].median()

print(f'The mean average of the order is ${mean_order}')
print(f'The median average of the order is ${median_order}')

The mean average of the order is $3145.128
The median average of the order is $284.0



When calculating the AOV using the mean, we come out to the same value of $3,145.13 previously calculated. This was the error in the calculation; as mentioned before, the outliers in the data caused the incorrect value.

## Question B & C
The preferred method for this calculation should be to use the median, which is not affected by outliers. When using this method, the value comes out to $284, which seems more reasonable given the context.